main.py

```diff
@@ -0,0 +1,124 @@
+
+
+def schoolsummary(surl, schoolpostaldistrict):
+
+#Extract the school name and address, concatenate with the postal district to compound the school summary information
+
+    urlschoolsumm = urlinit+surl #creating the school summary url
+    urlschoolsummget = requests.get(urlschoolsumm) # request the web page
+    urlschoolsummcontent = urlschoolsummget.content #extract the content of the web page in bytes
+    htmlschsummcontent= html.document_fromstring(urlschoolsummcontent) # HTML parsing
+
+    schoolname = htmlschsummcontent.xpath('//*[@id="area1"]/h1/text()') #extract school name
+    schooladdress = htmlschsummcontent.xpath('//*[@id="area1"]/p/text()[2]') #extract completed school address
+    #schoolenrollm = htmlschsummcontent.xpath('//*[@id="area1"]/p/text()[15]') #extract the school enrollment
+    #schoolfees = htmlschsummcontent.xpath('//*[@id="area1"]/p/text()[20]') #extract the scholl fee : private/public
+
+
+    #schoolgenre = re.findall(r'([A-Z]\w+):',schoolenrollm[0]) # Extract only words : Boys, Girls or Boys and Girls
+    #schooltype = re.findall(r'(Yes|No)\s',schoolfees[0]) # Extract only Yes or No
+    schooladdress[0] = schooladdress[0].strip() # Remove \n
+
+    schoolsummarylist = [schoolname[0], schooladdress[0], schoolpostaldistrict] #schoolgenre[0], schooltype[0]] #list of summary school
+
+    return schoolsummarylist
+
+
+def schoolprog(surl,schoolsummarylist):
+
+#Collect the content about each school progression.
+
+    surl=surl.replace('/school', '')
+    urlschoolproginit = urlinit+'/sch/'+surl+'/college-progression' #creating the college progression url
+    urlschoolprog = requests.get(urlschoolproginit) # request the web page
+    contenturlschoolprog = urlschoolprog.content # extract the content of the web page in bytes
+    htmlcontent = html.document_fromstring(contenturlschoolprog) #HMTL parsing
+    htmlelemprog = htmlcontent.cssselect("[valign=top]") # select all the block of data with the tag valign = top.
+
+
+    for elem in htmlelemprog: #collect the items on htmlelem list and transform to a list of strings
+        contentprog = elem.text_content()
+
+    splitcontentyear = contentprog.split('\n') #dividing the content into lines by year
+    splitcontentyear[0] = splitcontentyear[0].lstrip('Leaving Cert & College Numbers') # removing this text from the first line
+    splitcontentyear.remove(splitcontentyear[len(splitcontentyear)-1]) #removing the blank line
+
+    if len(splitcontentyear) == 0: # if there isnt school progression available for a school, fill with " ".
+        schoolproglist = [' ', ' ', ' ']
+        writecsv(schoolsummarylist, schoolproglist)
+    else:
+        splitcontentyear.remove(splitcontentyear[len(splitcontentyear)-1]) #removing a content on the last element not needed
+
+        for i in splitcontentyear:
+        #extract the year, total sitting leaving and total of main colleges from each year/each college/each region
```

Comment:
if variable `i` has meaningful data, i.e. it isn't just a counter.. you should name id with a more reasonable name. I don't know, maybe `contentyear`?

Reply:
ok